packages/x-components/src/x-modules/identifier-results/components/identifier-result.vue

```diff
@@ -9,64 +9,71 @@
 
 <script lang="ts">
   import { Result } from '@empathyco/x-types';
-  import Vue from 'vue';
-  import { Component, Prop } from 'vue-property-decorator';
-  import { Getter, State } from '../../../components/decorators/store.decorators';
-  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { computed, defineComponent, PropType } from 'vue';
   import { identifierResultsXModule } from '../x-module';
+  import { useGetter, useState } from '../../../composables';
 
   /**
    * This component renders an identifier result value and highlights its matching part with the
    * query from the state. Receives as prop the {@link @empathyco/x-types#Result} data.
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(identifierResultsXModule)]
-  })
-  export default class IdentifierResult extends Vue {
-    /**
-     * (Required) The {@link @empathyco/x-types#Result} information.
-     *
-     * @public
-     */
-    @Prop({ required: true })
-    protected result!: Result;
+  export default defineComponent({
+    name: 'IdentifierResult',
+    xModule: identifierResultsXModule.name,
+    props: {
+      /**
+       * (Required) The {@link @empathyco/x-types#Result} information.
+       *
+       * @public
+       */
+      result: {
+        type: Object as PropType<Result>,
+        required: true
+      }
+    },
+    setup: function (props) {
+      /**
+       * Query from the module state.
+       *
+       * @public
+       */
+      const { query } = useState('identifierResults', ['query']);
 
-    /**
-     * Query from the module state.
-     *
-     * @public
-     */
-    @State('identifierResults', 'query')
-    public query!: string;
+      /**
+       * The RegExp with the current query from the state adding the separatorChars after each
+       * matching character.
+       *
+       * @public
+       */
+      const { identifierHighlightRegexp } = useGetter('identifierResults', [
+        'identifierHighlightRegexp'
+      ]);
 
-    /**
-     * The RegExp with the current query from the state adding the separatorChars after each
-     * matching character.
-     *
-     * @public
-     */
-    @Getter('identifierResults', 'identifierHighlightRegexp')
-    public identifierHighlightRegexp!: RegExp;
+      /**
+       * Highlights the matching part of the identifier result with the query from the state.
+       *
+       * @returns String - The identifier result s query with the matching part inside a `<span>` tag.
+       * @public
+       */
+      const highlightedQueryHTML = computed((): string => {
+        const identifierValue = props.result.identifier?.value ?? '';
+        if (identifierValue && identifierHighlightRegexp.value) {
+          return identifierValue.replace(
+            identifierHighlightRegexp.value,
```

Comment:
Try to fix this warning

![Screenshot 2024-06-17 at 16 02 44](https://github.com/empathyco/x/assets/114984466/31de3e4b-b5b1-4358-a292-ec3ddcab446b)

Maybe you should assign `RegExp` type